src/meta/src/hummock/manager/checkpoint.rs

```diff
@@ -30,13 +30,20 @@ use risingwave_pb::hummock::{PbHummockVersionArchive, PbHummockVersionCheckpoint
 use thiserror_ext::AsReport;
 
 use crate::hummock::error::Result;
+use crate::hummock::manager::versioning::Versioning;
 use crate::hummock::manager::{read_lock, write_lock};
 use crate::hummock::metrics_utils::trigger_gc_stat;
 use crate::hummock::HummockManager;
 
 #[derive(Default)]
 pub struct HummockVersionCheckpoint {
     pub version: HummockVersion,
+
+    /// stale objects of versions before the current checkpoint.
+    ///
+    /// Previously we stored the stale object of each single version.
+    /// Currently we will merge the stale object between two checkpoints, and only the
+    /// id of the checkpointed hummock version are included in the map.
```

Comment:
Is this backward compatible?

Reply:
Yes, it's backward compatible. The existing stale objects was maintained per version. We can see it as a special case that creates a checkpoint for each version.